### PR TITLE
OS-8054 inotify watches lead to EBUSY during zfs mount

### PR DESCRIPTION
**Pull request by jasonbking**
Portions contributed by: Mike Gerdts <mike.gerdts@joyent.com>

### CODE REVIEW

File: usr/src/uts/common/sys/vnode.h [hunk 1]
```diff
@@ -285,6 +287,7 @@ typedef struct vnode {
 	kmutex_t	v_lock;		/* protects vnode fields */
 	uint_t		v_flag;		/* vnode flags (see below) */
 	uint_t		v_count;	/* reference count */
+	uint_t		v_phantom_count;
```
Comment (danmcd) [inline usr/src/uts/common/sys/vnode.h:290]: Please add a comment pointing below.

File: usr/src/uts/common/fs/vnode.c [hunk 2]
```diff
@@ -852,6 +852,31 @@ vn_rele(vnode_t *vp)
...
+vn_phantom_rele(vnode_t *vp)
+{
+	VERIFY(vp->v_count > 0);
+
+	mutex_enter(&vp->v_lock);
+	VERIFY3U(vp->v_count, >=, vp->v_phantom_count);
+	vp->v_phantom_count--;
+	DTRACE_PROBE1(vn__phantom_rele, vnode_t *, vp);
+	if (vp->v_count == 1) {
+		mutex_exit(&vp->v_lock);
```
Comment (danmcd) [inline usr/src/uts/common/fs/vnode.c:865]: CAN/SHOULD we `ASSERT0(vp->v_phantom_count);` prior to dropping the mutex?  IMHO it makes for nicer readability.

File: usr/src/uts/common/fs/vnode.c [hunk 3]
```diff
@@ -852,6 +852,31 @@ vn_rele(vnode_t *vp)
...
+		VOP_INACTIVE(vp, CRED(), NULL);
+		return;
+	}
+	VN_RELE_LOCKED(vp);
+	mutex_exit(&vp->v_lock);
+}
+
+uint_t
```
Comment (danmcd) [inline usr/src/uts/common/fs/vnode.c:873]: Comment explaining why we need this, please? I know the vnode.h really explains it, but someone passing by here might think, "WTF?"

File: usr/src/uts/common/fs/vnode.c (hunk 2 above)
Comment (jasonbking) [reply usr/src/uts/common/fs/vnode.c:865]: Yeah, I can add that.

File: usr/src/uts/common/sys/vnode.h (hunk 1 above)
Comment (jasonbking) [reply usr/src/uts/common/sys/vnode.h:290]: Done

File: usr/src/uts/common/fs/vnode.c (hunk 3 above)
Comment (jasonbking) [reply usr/src/uts/common/fs/vnode.c:873]: Done.

**Comment by jasonbking** [comment]
The intention was to get LX fixed first (many/most Linux distros end up watching mount points, which can break using delegated datasets in LX)

**Pull request merged**